src/sources/csv.rs

```diff
@@ -0,0 +1,261 @@
+//! CSV source for syncing with Famedly's Zitadel.
+
+use std::{fs, path::PathBuf};
+
+use anyhow::{Context, Result};
+use async_trait::async_trait;
+use csv::Reader;
+use serde::Deserialize;
+
+use super::Source;
+use crate::{user::User, zitadel::SourceDiff};
+
+/// CSV Source
+pub struct CsvSource {
+	/// CSV Source configuration
+	csv_config: CsvSourceConfig,
+}
+
+#[async_trait]
+impl Source for CsvSource {
+	fn get_name(&self) -> &'static str {
+		"CSV"
+	}
+
+	async fn get_diff(&self) -> Result<SourceDiff> {
+		let new_users = self.read_csv()?;
+		return Ok(SourceDiff { new_users, changed_users: vec![], deleted_user_ids: vec![] });
+	}
+}
+
+impl CsvSource {
+	/// Create a new CSV source
+	pub fn new(csv_config: CsvSourceConfig) -> Self {
+		Self { csv_config }
+	}
+
+	/// Get list of users from CSV file
+	fn read_csv(&self) -> Result<Vec<User>> {
+		let file_path = &self.csv_config.file_path;
+		let file = fs::File::open(&self.csv_config.file_path)
+			.context(format!("Failed to open CSV file {}", file_path.to_string_lossy()))?;
+		let mut reader = Reader::from_reader(file);
+		let users: Result<Vec<CsvData>, _> = reader.deserialize().collect();
+		let users: Result<Vec<User>, _> = users
+			.context("Failed to deserialize CSV data")
+			.map(|users| users.into_iter().map(CsvData::to_user).collect());
+		users
```

Comment:
Small suggestion to avoid unnecessary bindings:
```rust
		Ok(reader
			.deserialize()
			.collect::<Result<Vec<_>, _>>()
			.context("Failed to deserialize CSV data")?
			.into_iter()
			.map(CsvData::to_user)
			.collect())
```
Or to actually ignore deserialization errors:
```rust
		Ok(reader
			.deserialize()
			.map(|r| r.inspect_err(|x| tracing::error!("Failed to deserialize: {x}")))
			.filter_map(Result::ok)
			.map(CsvData::to_user)
			.collect())
```
Or if you still want to keep it splitted, we should unwrap IMO instead of `.map` on `Result`:
```rust
		let users: Vec<CsvData> = reader
			.deserialize()
			.collect::<Result<Vec<_>, _>>()
			.context("Failed to deserialize CSV data")?;
		Ok(users.into_iter().map(CsvData::to_user).collect())
```